app/src/shared/components/Globals.tsx

```diff
@@ -0,0 +1,14 @@
+import { useIsPersistingAnyStreamDraft } from '$shared/stores/streamEditor'
+import usePreventNavigatingAway from '../hooks/usePreventNavigatingAway'
+
+export default function Globals() {
+    const isPersistingAnyStreamDraft = useIsPersistingAnyStreamDraft()
+
+    usePreventNavigatingAway({
+        isDirty(destination) {
+            return typeof destination === 'undefined' && isPersistingAnyStreamDraft
+        }
+    })
```

Comment:
@pmichalowski, that's where the `useIsPersistingAnyStreamDraft` is useful. In short, if any draft is being persisted we have a global `usePreventNavigatingAway` that'll intervene even if the user is some place else.